pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -453,9 +464,19 @@ func updateConfigSpec(
 	configSpec := &vimTypes.VirtualMachineConfigSpec{}
 	vmClassSpec := updateArgs.VMClass.Spec
 
-	UpdateHardwareConfigSpec(config, configSpec, &vmClassSpec)
-	UpdateConfigSpecCPUAllocation(config, configSpec, &vmClassSpec, updateArgs.MinCPUFreq)
-	UpdateConfigSpecMemoryAllocation(config, configSpec, &vmClassSpec)
+	// Before VM Class as Config, VMs were deployed from the OVA, and are then
+	// reconfigured to match the desired CPU and memory reservation.  Maintain that
+	// behavior.  With the FSS enabled, VMs will be _created_ with desired HW spec, and we
+	// will not modify the hardware of the VM post creation.  So, don't populate the
+	// Hardware config and CPU/Memory reservation.
+	if !lib.IsVMClassAsConfigFSSDaynDateEnabled() {
```

Comment:
I believe that would leave us with a window where a CL Deploy is running and then a upgrade is kicked off, and the first VM "update" reconcile runs with the FFS enabled.

Reply:
The chances of this happening are so slim that it is okay to ignore in favor of _not_ reconciling each VM with update hardware spec/reservations during power cycle.  Would you agree?